src/lsm/compaction.zig

```diff
@@ -315,6 +317,25 @@ pub fn CompactionType(
             assert(compaction.io_pending == 0);
             assert(!compaction.merge_done);
 
+            // Move table optimization
+            if (compaction.grid_reservation == null) {
+                const snapshot_max = snapshot_max_for_table_input(compaction.op_min);
+                const level_b = compaction.level_b;
+                const level_a = level_b - 1;
+
+                var table_a = compaction.level_a_input.?;
+                compaction.manifest.update_table(level_a, snapshot_max, &table_a);
+                assert(table_a.snapshot_max == snapshot_max);
+
+                table_a = compaction.level_a_input.?;
+                compaction.manifest.insert_table(level_b, &table_a);
```

Comment:
Good point, it doesn't. Could we use `TableInfo.flags` to denote "moved" instead of "removed" when purging invisible tables? Thinking of implementing move_table entirely in Manifest.

Reply:
That, or remove the table right then and there (inside Manifest) instead of waiting for `remove_invisible_tables` at the end of Compaction.

Reply:
Implementing `move_table` in `Manifest` makes sense to me (in fact I think that's where it originally lived).


> That, or remove the table right then and there (inside Manifest) instead of waiting for remove_invisible_tables at the end of Compaction.

There is a comment in `tree.zig`:

```
            // All compactions have finished for the current half-bar.
            // We couldn't remove the (invisible) input tables until now because prefetch()
            // needs a complete set of tables for lookups to avoid missing data.
```

i.e. if you move the table during beat number 1, we will keep querying the table (in its original location/level) in the next beat.

Reply:
I think a table flag is a decent solution, though it is a little weird since the table flags get persisted, but this particular flag would never be persisted.

Reply:
Instead of flags, Im thinking of using a separate `[constants.lsm_levels]usize` buffer of `TableInfo.address`